django_mongodb/query.py

```diff
@@ -0,0 +1,372 @@
+import re
+from functools import wraps
+
+from django.core.exceptions import FullResultSet
+from django.db import DatabaseError, IntegrityError, NotSupportedError
+from django.db.models.lookups import UUIDTextMixin
+from django.db.models.query import QuerySet
+from django.db.models.sql.where import OR, SubqueryConstraint
+from django.utils.tree import Node
+from pymongo import ASCENDING, DESCENDING
+from pymongo.errors import DuplicateKeyError, PyMongoError
+
+
+def safe_call(func):
+    @wraps(func)
+    def wrapper(*args, **kwargs):
+        try:
+            return func(*args, **kwargs)
+        except DuplicateKeyError as e:
+            raise IntegrityError from e
+        except PyMongoError as e:
+            raise DatabaseError from e
+
+    return wrapper
+
+
+def safe_regex(regex, *re_args, **re_kwargs):
+    def wrapper(value):
+        return re.compile(regex % re.escape(value), *re_args, **re_kwargs)
+
+    wrapper.__name__ = "safe_regex (%r)" % regex
+
+    return wrapper
+
+
+OPERATORS_MAP = {
+    "exact": lambda val: val,
+    "gt": lambda val: {"$gt": val},
+    "gte": lambda val: {"$gte": val},
+    "lt": lambda val: {"$lt": val},
+    "lte": lambda val: {"$lte": val},
+    "in": lambda val: {"$in": val},
+    "range": lambda val: {"$gte": val[0], "$lte": val[1]},
+    "isnull": lambda val: None if val else {"$ne": None},
```

Comment:
should this be `{"$eq": None}`?

Reply:
As far as I tested, the queries `{'field': None}` and `{'field': {'$eq': None}}` give the same results. Is that your concern?

Reply:
Oh, I misread the output. Yes, that's fine. 

So showing `{"$ne": None}` probably occurs when the query evaluation is something like...
`Models.object.filter(field__isnull=False)`